tools/killsnoop.bt

```diff
@@ -47,9 +88,20 @@ tracepoint:syscalls:sys_exit_tkill,
 tracepoint:syscalls:sys_exit_tgkill
 /@tpid[tid]/
 {
-	printf("%-15s %8d %-16s %4d %8d %6d\n",
-		strftime("%H:%M:%S.%f", nsecs),
-		pid, comm, @tsig[tid], @tpid[tid], args.ret);
+	printf("%-15s %8d %-16s", strftime("%H:%M:%S.%f", nsecs), pid, comm);
+	if (@SIG_STR[@tsig[tid]] == "") {
```

Comment:
```suggestion
	if (!has_key(@SIG_STR, @tsig[tid])) {
```

Reply:
Thinking out loud, we had discussed previously improving the options around maps to allow you to use the different lookup semantics effectively. Go has two forms of this, `x = m[...]` or `x, ok = m[...]` where `ok` is a boolean that indicates whether the value was there or not. (And x would be the default zero value in that case.) That gets lowered to different intrinsics, depending on which is used.

Obviously this is a nice improvement as is, but it would be cool if we had:

```
$name, $exists = @SIG_STR[@tsig[tid]];
...
```

Reply:
@jordalgo Use `has_key`, thanks. @amscanne Thanks.

Reply:
@amscanne That's not bad but feels a bit premature for bpftrace. We'd discussed trying to keep the syntax as simple as possible (at least in the near term) and that feel a bit much. I do like that it would save a map lookup call.

On an even separate tangent, this PR got me thinking about `const`s and how we can save hash map lookups if we converted `@SIG_STR` to a global object of some sort or even a per-cpu hash.